docs/nixops-krops-migration.md

```diff
@@ -0,0 +1,101 @@
+# Tutorial: Moving from a NixOps to a Krops deployment
+
+1. Add a new ssh key to your nix-bitcoin node
+
+   Krops doesn't automatically generate ssh keys like NixOps, instead you add your own.
+
+   If you don't have a ssh key yet
+
+   ```
+   ssh-keygen -t ed25519 -f ~/.ssh/bitcoin-node
+   ```
+
+   Edit `configuration.nix`
+
+   ```
+   users.users.root = {
+     openssh.authorizedKeys.keys = [
+       "<contents of ~/.ssh/bitcoin-node.pub or existing .pub key file>"
+     ];
+   };
+   ```
+
+   Deploy new key
+
+   ```
+   nixops deploy -d bitcoin-node
+   ```
+
+2. Update your nix-bitcoin, depending on your setup either with `fetch-release` or `git`. Make sure you are at least on `v0.0.41`.
+
+3. Pull the latest nix-bitcoin source
+
+    ```
+    cd ~/nix-bitcoin
+    git pull
+    ```
+
+4. Copy new and updated files into your deployment folder
+
+    ```
+    cd <deployment directory, for example `~/nix-bitcoin-node`>
+    cp -r ~/nix-bitcoin/examples/{krops,krops-configuration.nix,shell.nix} .
```

Comment:
Yes, I thought about this, but it requires a symlink to `configuration.nix` in `krops/` which might be confusing.
[Here's an implementation](https://github.com/erikarvstedt/nix-bitcoin/commit/590438e). I'll let you decide what to use.

Reply:
How about this https://github.com/jonasnick/nix-bitcoin/commit/3693f0b8f6b4bf77136be9712e00130dc3307b0d. I haven't tested this with `krops-deploy` and it seems a bit contrived. I think we may also be able to remove krops-configuration.nix altogether and import the nix-bitcoin krops module (as well as `configuration.nix`) in `nixos-config = builtins.toFile nixos-config ''` directly.

Reply:
Note to self: would also need to update install and migration instructions.

Reply:
Compared to 3693f0b8f6b4bf77136be9712e00130dc3307b0d I'd prefer the symlink because it's much simpler. It also avoids the otherwise empty `krops` dir on the target node.

Reply:
What do you mean, `krops` contains `krops-configuration.nix`, both with deploy-krops.sh and in real deployments.

Reply:
That's what I meant by `otherwise empty`. It only contains a single file, so it's redundant.

Reply:
Note that with https://github.com/fort-nix/nix-bitcoin/commit/3693f0b8f6b4bf77136be9712e00130dc3307b0d you have to duplicate that elaborate krops file filter to also move `krops-vm-configuration.nix` into `krops/`.

Reply:
Ah I see what you mean with "otherwise empty". The upside is that it replicates the local directory structure. I don't see why we'd have to duplicate the filter for krops-vm-configuration.nix. `deploy-krops.sh` works just fine.

Reply:
I meant: only
> when we also move krops-vm-configuration.nix into krops/.

Reply:
*I don't see why we'd have to move `krops-vm-configuration.nix` into `krops/`

Reply:
Ah, ok, you wanted to only move krops files into `krops` that are relevant for setting up a real node.

Reply:
If you like to go with the filter solution, you can simplify the `nix-config` definition to
`nix-config.symlink = "krops/krops-configuration.nix";`

Reply:
That's a neat trick and I think that makes it the (slightly) more preferable solution overall then.

Reply:
Agree?

Reply:
If you want `krops/` to only include files relevant for setting up a real node, the following is the optimal solution:
https://github.com/erikarvstedt/nix-bitcoin/commit/d34bd3e885481ef9e6be1a160157adee9b80078a

Note that `krops/krops-configuration.nix` is not used by the examples or the real deploy script, it only allows users to build/eval their config locally.

Reply:
Ah. Being able to build the config locally is actually quite useful because syncing the files is so slow. I was already thinking about a PR to add some alias to the nix-shell that does that.

Reply:
ACK fwiw